r/tests/testthat/test-codec-availability.R

```diff
@@ -0,0 +1,24 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+context("CodecAvailability")
+
+test_that("codec_is_available", {
+  expect_true(codec_is_available("lz4"))
+  expect_true(codec_is_available("zstd"))
+})
```

Comment:
Thanks for adding this. A few points:

* I think you could put this under `test-compressed.R` where there are other codec tests
* Change the test name (in `test_that()` to something like "Compression codecs are included in the Windows build" because that's what we're testing, not that `codec_is_available` works
* Wrap the assertions (or even the whole `test_that()`) in `if (identical(Sys.getenv("APPVEYOR"), "True")) { ... }` (i.e. the inverse of what `testthat::skip_on_appveyor()` does) because we're testing that that Windows build is including the right stuff. It's not so much a test of the code as of the build system.